### PR TITLE
Feature 4167 Schaff Trend Cycle indicator

### LINKED ISSUE #4167: Schaff Trend Cycle Indicator
#### Expected Behavior
Support "Schaff Trend Cycle Indicator"
https://www.investopedia.com/articles/forex/10/schaff-trend-cycle-indicator.asp

#### Actual Behavior
No support for "Schaff Trend Cycle Indicator"

#### Potential Solution
Implement "Schaff Trend Cycle Indicator". Add unit tests

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Addition of a new indicator: Schaff Trend Cycle

#### Related Issue

Closes #4167 

#### Motivation and Context

Add a new indicator that builds on previous ones. STC takes MACD further and is more responsive to momentum changes.

#### Requires Documentation Change

Yes, addition of this new indicator to the indicator documentation is required.

#### How Has This Been Tested?

External data has been added in this PR to run against the indicator using the CommonIndicatorTests. Reference Tests/Indicators/SchaffTrendCycleTests.cs for test setup.

Also in development I worked through the formula using an excel to compare it to a TradingView script's calculation to ensure accuracy of the algorithm. This particular indicator is extremely sensitive as it chains EMAs and Stochastics on top of a MACD calculation so the more data points processed the closer the values become. In the case of TradingView it was able to process data from before the start which leads the beginning STC values to differ greatly in the excel. You will notice after row 120 the variance becomes very minimal.

[spy_stc_excel_vs_TV.txt](https://github.com/QuantConnect/Lean/files/4866844/spy_stc_excel_vs_TV.txt)

The test file I included (spy_stc.txt) in this PR goes even further with over 1700 points for an intraday comparison. Running my indicator against this show a very small max +/- 1 variance between the values. To reproduce run SchaffTrendCycleTests.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by C-SELLERS** [comment]
Looks like when it builds my test data in [Tests/TestData/spy_stc.txt](https://github.com/QuantConnect/Lean/pull/4557/commits/1d1a68a78bae3d0a8e9f1c0dbb54408176fbb995) is not being transferred to the Release test bin. 

See Travis build error [here](https://travis-ci.org/github/QuantConnect/Lean/builds/704477343#L29249) 

Can whoever reviews this let me know how to get the test data into the correct directory?

**Comment by C-SELLERS** [comment]
Closing to restart the test. Failed on something unrelated.

File: Indicators/readme.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
+<h2>Guide to Developing Indicators in Lean</h2>
```
Comment (jaredbroad) [inline Indicators/readme.md:1]: Please remove the doc from this PR and we can draft offline and submit to the wiki or the tutorials repo.

Comment (C-SELLERS) [reply Indicators/readme.md]: Removed!

**Comment by jaredbroad** [comment]
We forgot one thing! The shortcut helper in QCAlgorithm.Indicators.cs 

`STC(int cyclePeriod, int fastPeriod, int slowPeriod, MovingAverageType type)`

**Comment by C-SELLERS** [comment]
Uh oh, Ill take care of it!

**Pull request merged**